src/Frontend/Core.hs

```diff
@@ -191,7 +191,7 @@ class Page p => FormPage p where
 
 -- | Defines some properties for pages
 class Page p where
-    isPrivatePage :: p -> Bool
+    isPrivatePage :: proxy p -> Bool
```

Comment:
This is strictly more general than `Proxy` since `isPrivatePage (Proxy :: Proxy a)` will still work. So is `isPrivatePage [a]` (where `a` is a value here). So is `isPrivatePage mp` where `mp` is of type `m p`.